### ISSUE TITLE
CPU_Tensor_Network overfitting to data points

### ISSUE THREAD

**Issue by bnfrsMusic** [bnfrsMusic/MLRust#1]
### Description

The current algorithm seems to be overfitting to every new data point. 

---

### Example

When using the following Vecs as the inputs and targets, with a 2x4x4x1 MLP, the MLP is able to train on it. However, it changes all the values to match that result **only**. 

**Code:**
```
    let input_arr: Vec<Tensor> = vec![
        Tensor::from(vec![2, 1], vec![1.0, 0.0]),
        Tensor::from(vec![2, 1], vec![0.0, 1.0]),
        Tensor::from(vec![2, 1], vec![1.0, 1.0]),
    ];
    let target_arr: Vec<Tensor> = vec![
        Tensor::from(vec![1, 1], vec![0.0]),
        Tensor::from(vec![1, 1], vec![1.0]),
        Tensor::from(vec![1, 1], vec![0.0]),
    ];
```

**Result:**
Varies slightly every run due to the randomization of the tensors. Here is the result of each data point after their respective training (100 epochs for each data point, 0.05 learning rate):

```
Output 0: [0.042997837012440455]
Output 1: [0.9448928448976363]
Output 2: [0.04791337099573982]
```

While those seem correct, if you were to get the result for each after all three trainings, you would get the following:
```
Output 0: [0.04798416359063825]
Output 1: [0.04739009704330084]
Output 2: [0.04791337099573982]
```
This shows that it is taking the target data point as the target for all values, and thus fitting to that value only.
